### PR TITLE
[Smartswitch] Yang model change for bridge midplane

### PR DESCRIPTION
**Pull request by gpunathilell**
#### Why I did it
The yang model for bridge-midplane is not aligned with the actual implementation present in https://github.com/sonic-net/sonic-buildimage/pull/18178, 
This PR is to align the yang model to change `bridge_midplane` to `bridge-midplane` since this is the value used in the `systemd-networkd` in CONFIG_DB so that the yang validation does not fail

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How to verify it
Enhance unit test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by gpunathilell** [comment]
/azpw.run Azure.buildimage

**Review by oleksandrivantsiv — APPROVED** [review]

File: src/sonic-dhcp-utilities/dhcp_utilities/dhcpservd/dhcp_cfggen.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def _parse_dpu(self, dpus_table, mid_plane_table):
         Returns:
             Parsed obj, sample:
                 mid_plane = {
-                    "bridge": "bridge_midplane",
+                    "bridge": "bridge-midplane",
```
Comment (qiluo-msft) [inline src/sonic-dhcp-utilities/dhcp_utilities/dhcpservd/dhcp_cfggen.py:110]: Is it a valid requirement for the alignment? Does any guidance mentioned the naming convention?

Comment (dgsudharsan) [reply src/sonic-dhcp-utilities/dhcp_utilities/dhcpservd/dhcp_cfggen.py:110]: @qiluo-msft It is as per the HLD. Please refer to https://github.com/sonic-net/SONiC/pull/1586/files

**Review by ganglyu — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**